rules/naming/src/ExpectedNameResolver/AbstractExpectedNameResolver.php

```diff
@@ -17,40 +16,32 @@ abstract class AbstractExpectedNameResolver implements ExpectedNameResolverInter
      */
     protected $nodeTypeResolver;
 
-    /**
-     * @var PropertyNaming
-     */
-    protected $propertyNaming;
-
     /**
      * @var NodeNameResolver
      */
     protected $nodeNameResolver;
 
-    public function __construct(
-        NodeNameResolver $nodeNameResolver,
-        NodeTypeResolver $nodeTypeResolver,
-        PropertyNaming $propertyNaming
-    ) {
+    public function __construct(NodeNameResolver $nodeNameResolver, NodeTypeResolver $nodeTypeResolver)
+    {
         $this->nodeNameResolver = $nodeNameResolver;
         $this->nodeTypeResolver = $nodeTypeResolver;
-        $this->propertyNaming = $propertyNaming;
     }
 
-    public function resolveIfNotYet(Property $property): ?string
+    public function resolveIfNotYet(Node $node): ?string
```

Comment:
Here should be some Node type specification in `@param`

```suggestion
    /**
     * @name Property|Variable|...
     */
    public function resolveIfNotYet(Node $node): ?string
```

So the scope is narrow and it's easier to debug later. Now it can be anything, even `Switch_` :D 

Also these `Property|Variable|...` are [PHP 8-ready union types](https://wiki.php.net/rfc/union_types_v2). So we can upgrade Rector this year to:

```php
    public function resolveIfNotYet(Property|Variable Node $node): ?string
```

Reply:
For now it is just Property and Param.

Reply:
These should be there.

Reply:
I think also in the contract, to make the whole family of contract + implementer type aware

Reply:
Now, it should be okay.
Interface and abstract class have Property|Param.
Name resolvers have appropriate Node type(s) depending on the Node type(s) they handle.

Reply:
Thank you